ledger/src/shred/payload.rs

```diff
@@ -3,12 +3,18 @@ use std::{
     sync::Arc,
 };
 
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq)]
 pub enum Payload {
     Shared(Arc<Vec<u8>>),
     Unique(Vec<u8>),
 }
 
+enum Bytes {
+    A(Vec<u8>),
+    B([u8; 1203]), // Merkle data shreds.
+    C([u8; 1228]), // Merkle coding shreds, and legacy shreds.
```

Comment:
Won't these array variants make any instance of the enum at least 1128 bytes, even the `A` case which needs much less (24 bytes for Vec + whatever for the enum & alignment) ?

Reply:
the `Vec` case will only be used for legacy shreds (so basically never outside of tests).

Reply:
Ahh ok, basically a non-issue then.

Semi-related - are you thinking about ripping legacy shred code/support out ?

Reply:
> are you thinking about ripping legacy shred code/support out ?

we should. but right now I don't have the time to patch all tests which are not compatible with merkle shreds.

Reply:
Good to know